src/solvers/flattening/boolbv_index.cpp

```diff
@@ -200,10 +199,7 @@ bvt boolbvt::convert_index(const index_exprt &expr)
 
     // get literals for the whole array
 
-    const bvt &array_bv=convert_bv(array);
-
-    if(array_size*width!=array_bv.size())
-      throw "unexpected array size";
+    const bvt &array_bv = convert_bv(array, array_size.to_ulong() * width);
```

Comment:
Please use `numeric_cast_v<std::size_t>(array_size * width)` as it will 1) make sure there is no risk for overflow in the multiplication, 2) combine the check and the conversion, and 3) hide the implementation detail that `bigint`/`.to_ulong` is.